File: patterns/lichtsteuerungssystem/rules-js/lichtsteuerungssystem-bewegung.js
rules.JSRule({
  name: "Motion detected",
  triggers: [triggers.ItemStateUpdateTrigger("MotionSensor1", "ON")],
  execute: (event) => {
    if (items.getItem("Light1").state === "OFF") {
      items.getItem("Light1").sendCommand("ON");
      console.log("Bewegung erkannt - Light1 eingeschaltet (10 Min. Timer)");
      actions.ScriptExecution.createTimer(time.ZonedDateTime.now().plusMinutes(10), () => {
        items.getItem("Light1").sendCommand("OFF");
      });
    } else {
      console.log("Bewegung erkannt, Light1 war aber bereits an");
    }
  }
});

File: patterns/lichtsteuerungssystem/rules-python/lichtsteuerungssystem_bewegung.py
import threading

from openhab import rule, Registry
from openhab.triggers import ItemStateUpdateTrigger


@rule(triggers=[ItemStateUpdateTrigger("MotionSensor1", "ON")])
class MotionDetected:
    def execute(self, module, input):
        if str(Registry.getItem("Light1").getState()) == "OFF":
            Registry.getItem("Light1").sendCommand("ON")
            self.logger.info("Bewegung erkannt - Light1 eingeschaltet (10 Min. Timer)")

            def turn_off():
                Registry.getItem("Light1").sendCommand("OFF")

            threading.Timer(10 * 60, turn_off).start()
        else:
            self.logger.info("Bewegung erkannt, Light1 war aber bereits an")
